subjects/06-apps/exercise.js

```diff
@@ -12,6 +12,157 @@
 var React = require('react');
 var { login, sendMessage, subscribeToMessages } = require('./utils/ChatUtils');
 
+// you need to npm install this!!!!!
+var caf = require('cool-ascii-faces');
+
+var ChatApp = React.createClass({
+  getInitialState() {
+    return {
+      messages: []
+    }
+  },
+
+  componentDidMount() {
+    this.state.unsubscribe = this.subscribeToMessages(this.props.channel)
+  },
+
+  removeBadMessages(message) {
+    return !(message.text.length < 2 || message.text.length > 200)
```

Comment:
lorem ipsum etc etc etc; I likely missed things during the workshop by filtering these out.